File: questPresentation/src/main/java/ru/nekit/android/qls/quest/view/mediator/adapter/SquareItemAdapter.java
package ru.nekit.android.qls.quest.view.mediator.adapter;

import android.support.annotation.CallSuper;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

public class SquareItemAdapter<T extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<T> {

    private int mSize;

    public void setSize(int value) {
        mSize = value;
    }

    @Override
    public T onCreateViewHolder(ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    @CallSuper
    public void onBindViewHolder(T holder, int position) {
        ViewGroup.LayoutParams layoutParams = holder.itemView.getLayoutParams();
        layoutParams.height = mSize;
        layoutParams.width = mSize;
    }

    @Override
    public int getItemCount() {
        return 0;
    }

}

File: questPresentation/src/main/java/ru/nekit/android/qls/quest/view/mediator/adapter/SquareItemAdapter.kt
package ru.nekit.android.qls.quest.view.mediator.adapter

import android.support.annotation.CallSuper
import android.support.v7.widget.RecyclerView
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.subjects.Subject
import ru.nekit.android.qls.utils.IAutoDispose

//ver 1.0
abstract class SquareItemAdapter<T : RecyclerView.ViewHolder>(protected val answerPublisher: Subject<Any>) :
        RecyclerView.Adapter<T>(), IAutoDispose {

    override var disposable: CompositeDisposable = CompositeDisposable()

    var size: Int = 0

    @CallSuper
    override fun onBindViewHolder(holder: T, position: Int) {
        val layoutParams = holder.itemView.layoutParams
        layoutParams.height = size
        layoutParams.width = size
    }

    override fun onDetachedFromRecyclerView(recyclerView: RecyclerView) {
        dispose()
        super.onDetachedFromRecyclerView(recyclerView)
    }

    override fun getItemCount(): Int {
        return 0
    }

}